htdocs/fourn/class/fournisseur.product.class.php

```diff
@@ -1241,4 +1271,96 @@ private function logPrice($user, $datec, $buyprice, $qty, $multicurrency_buypric
 			return 1;
 		}
 	}
+
+	/**
+     * Load list of objects in memory from the database.
+     *
+     * @param  string      $sortorder    Sort Order
+     * @param  string      $sortfield    Sort field
+     * @param  int         $limit        limit
+     * @param  int         $offset       Offset
+     * @param  array       $filter       Filter array. Example array('field'=>'valueforlike', 'customurl'=>...)
+     * @param  string      $filtermode   Filter mode (AND or OR)
+     * @return array|int                 int <0 if KO, array of pages if OK
+     */
+    public function fetchAll($sortorder = '', $sortfield = '', $limit = 0, $offset = 0, array $filter = array(), $filtermode = 'AND')
```

Comment:
This method does not seems to be field of the class ProductFournisseur (related to table llx_product) but are related to table llx_product_fournisseur_price (so the table of Vendor Prices).
So this method must be available into a new class/file ProductFournisseurPrice

Reply:
@eldy Where do you prefer to put this CRUD class ? into htdocs/fourn/class or htdocs/product/class ?

Reply:
Good question.
Let's say into htdocs/product/class
I have a hope to see one day the fourn directory be removed and replaced into supplier_order and supplier_invoice.